### PR TITLE
Introduce periodic check

### PR DESCRIPTION
**Pull request by eolivelli**
- introduce metadata.refreshPeriodSeconds
- fix typo in metadata.refreshAtBoot (keep compatibility)
- upgrade HerdDB, ZooKeeper and BookKeeper deps
- use Curator testing
- add GitHub actions

### CODE REVIEW

**Comment by eolivelli** [comment]
All tests are passing locally
once we merge this patch GitHub Actions will kick in for new Pull Requests
I have copied the GitHub stuff from HerdDB recent work

@aluccaroni

File: web/src/test/java/org/bkvm/utils/AbstractBookkeeperTestUtils.java [hunk 1]
```diff
@@ -52,22 +52,19 @@
     @Rule
     public final TemporaryFolder folder = new TemporaryFolder();
 
-    LocalZookeeperServer zkServerMain;
```
Comment (mino181295) [inline web/src/test/java/org/bkvm/utils/AbstractBookkeeperTestUtils.java:55]: If we use Curator we should delete this file

File: .github/workflows/pr-validation.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        with:
+          java-version: '11'
+      - name: 'Cache Maven packages'
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: 'cache'
+          restore-keys: 'cache'
+      - name: 'Build with Maven'
+        run: mvn -B checkstyle:check apache-rat:check install spotbugs:check -Dmaven.test.redirectTestOutputToFile=true -Dherddb.file.requirefsync=false --file pom.xml
```
Comment (mino181295) [inline .github/workflows/pr-validation.yml:40]: Are we using apache-rat ?

**Review by mino181295 — CHANGES_REQUESTED** [review]

File: web/src/test/java/org/bkvm/utils/AbstractBookkeeperTestUtils.java (hunk 1 above)
Comment (eolivelli) [reply web/src/test/java/org/bkvm/utils/AbstractBookkeeperTestUtils.java:55]: good catch

File: .github/workflows/pr-validation.yml (hunk 2 above)
Comment (eolivelli) [reply .github/workflows/pr-validation.yml:40]: no, we aren't still using it.
Let's enable it in a second step.
We must add that tool one day

**Pull request merged**